src/goto-programs/remove_unused_functions.cpp

```diff
@@ -78,7 +78,10 @@ void find_used_functions(
           const code_function_callt &call = to_code_function_call(it->code);
 
           // check that this is actually a simple call
-          assert(call.function().id()==ID_symbol);
+          INVARIANT(
+            call.function().id() == ID_symbol,
+            "by this point, all kinds of function calls should have been "
+            "converted to simple function calls");
```

Comment:
Maybe add the error message as a comment in here, but really `to_symbol_expr` below actually takes care of this.